lib/pages/portal/view/portal_page.dart

```diff
@@ -76,51 +74,35 @@ class _PortalPageState extends State<PortalPage> {
   }
 
   Widget websiteCircle(Website website, double size) {
+    final bool canEdit = editingEnabled &&
+        (website.isPrivate || (user.canEditPublicWebsite ?? false));
     return Padding(
-      padding: const EdgeInsets.all(8),
-      child: FutureBuilder<ImageProvider<dynamic>>(
-        future: StorageProvider.imageFromPath(website.iconPath),
-        builder: (context, snapshot) {
-          ImageProvider image;
-          if (snapshot.hasData) {
-            image = snapshot.data;
-          } else {
-            image = AssetImage('assets/${website.iconPath}') ??
-                const AssetImage('assets/images/white.png');
-          }
-
-          final bool canEdit = editingEnabled &&
-              (website.isPrivate || (user.canEditPublicWebsite ?? false));
-          return CircleImage(
-            label: website.label,
-            tooltip: website.infoByLocale[LocaleProvider.localeString],
-            image: image,
-            enableOverlay: canEdit,
-            circleSize: size,
-            onTap: () {
-              if (canEdit) {
-                Navigator.of(context)
-                    .push(MaterialPageRoute<ChangeNotifierProvider>(
-                  builder: (_) => ChangeNotifierProvider<FilterProvider>(
-                    create: (_) => FilterProvider(
-                        defaultDegree: website.degree,
-                        defaultRelevance: website.relevance),
-                    child: WebsiteView(
-                      website: website,
-                      updateExisting: true,
-                    ),
+        padding: const EdgeInsets.all(8),
+        child: WebsiteIcon(
+          website: website,
+          canEdit: canEdit,
+          size: size,
+          onTap:() {
+            if (canEdit) {
+              // ignore: inference_failure_on_instance_creation
```

Comment:
Don't ignore warnings unless you specifically need that behaviour. You need to specify the type, so `MaterialPageRoute<ChangeNotifierProvider>(...)`.